provider/resource.go

```diff
@@ -108,6 +115,14 @@ type resource struct {
 
 var (
 	autogeneratedAWSResourcesRe = regexp.MustCompile(`^aws:(?:autoscaling|cloudformation)`)
+	providerResources           = map[string]func(string) (*tfdocs.Resource, error){
+		"aws": func(rt string) (*tfdocs.Resource, error) {
+			return awsdocs.GetResource(rt)
+		},
+		"google": func(rt string) (*tfdocs.Resource, error) {
+			return googledocs.GetResource(rt)
+		},
+	}
```

Comment:
Ok a better way:

* Define a function type `func(rt string) (*tfdocs.Resource, error)`
* Then the map just have to have `"google" googledocs.GetResource`

Which make is much more simpler (that was my first intention but IDK why I read it wrong on the implementation).

Or actually the way you have it now you can still directly use the format without defining the function actually :smile: